File: test/aidbox_sdk/snapshots/dotnethl7-fhir-r4-core/QuestionnaireResponse.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class QuestionnaireResponse : DomainResource, IResource
{
    public string? Questionnaire { get; set; }
    public Base.ResourceReference? Encounter { get; set; }
    public QuestionnaireResponse_Item[]? Item { get; set; }
    public Base.ResourceReference? Source { get; set; }
    public Base.ResourceReference? Author { get; set; }
    public required string Status { get; set; }
    public Base.Identifier? Identifier { get; set; }
    public Base.ResourceReference[]? BasedOn { get; set; }
    public string? Authored { get; set; }
    public Base.ResourceReference[]? PartOf { get; set; }
    public Base.ResourceReference? Subject { get; set; }

    public class QuestionnaireResponse_Item_Answer : BackboneElement
    {
        public Base.ResourceReference? ValueReference { get; set; }
        public string? ValueUri { get; set; }
        public string? ValueTime { get; set; }
        public string? ValueDecimal { get; set; }
        public Base.Quantity? ValueQuantity { get; set; }
        public object? Value    
        {
            get
            {
                if (ValueReference is not null)
                {
                    return ValueReference;
                }
        
                if (ValueUri is not null)
                {
                    return ValueUri;
                }
        
                if (ValueTime is not null)
                {
                    return ValueTime;
                }
        
                if (ValueDecimal is not null)
                {
                    return ValueDecimal;
                }
        
                if (ValueQuantity is not null)
                {
                    return ValueQuantity;
                }
        
                if (ValueString is not null)
                {
                    return ValueString;
                }
        
                if (ValueBoolean is not null)
                {
                    return ValueBoolean;
                }
        
                if (ValueDateTime is not null)
                {
                    return ValueDateTime;
                }
        
                if (ValueDate is not null)
                {
                    return ValueDate;
                }
        
                if (ValueCoding is not null)
                {
                    return ValueCoding;
                }
        
                if (ValueInteger is not null)
                {
                    return ValueInteger;
                }
        
                if (ValueAttachment is not null)
                {
                    return ValueAttachment;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(Base.ResourceReference))
                {
                    ValueReference = (Base.ResourceReference)value;
                    return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ValueUri = (string)value;
                    return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ValueTime = (string)value;
                    return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ValueDecimal = (string)value;
                    return;
                }
        
                if (value?.GetType() == typeof(Base.Quantity))
                {
                    ValueQuantity = (Base.Quantity)value;
                    return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ValueString = (string)value;
                    return;
                }
        
                if (value?.GetType() == typeof(bool))
                {
                    ValueBoolean = (bool)value;
                    return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ValueDateTime = (string)value;
                    return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ValueDate = (string)value;
                    return;
                }
        
                if (value?.GetType() == typeof(Base.Coding))
                {
                    ValueCoding = (Base.Coding)value;
                    return;
                }
        
                if (value?.GetType() == typeof(int))
                {
                    ValueInteger = (int)value;
                    return;
                }
        
                if (value?.GetType() == typeof(Base.Attachment))
                {
                    ValueAttachment = (Base.Attachment)value;
                    return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public Base.ResourceReference[]? Item { get; set; }
        public string? ValueString { get; set; }
        public bool? ValueBoolean { get; set; }
        public string? ValueDateTime { get; set; }
        public string? ValueDate { get; set; }
        public Base.Coding? ValueCoding { get; set; }
        public int? ValueInteger { get; set; }
        public Base.Attachment? ValueAttachment { get; set; }
    }

    public class QuestionnaireResponse_Item : BackboneElement
    {
        public Base.ResourceReference[]? Item { get; set; }
        public string? Text { get; set; }
        public QuestionnaireResponse_Item_Answer[]? Answer { get; set; }
        public required string LinkId { get; set; }
        public string? Definition { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/QuestionnaireResponse.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Attachment
from base import Coding
from base import Quantity
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class QuestionnaireResponseItemAnswer(BackboneElement):
    value_reference: Optional[Reference] = None
    value_uri: Optional[str] = None
    value_time: Optional[str] = None
    value_decimal: Optional[float] = None
    value_quantity: Optional[Quantity] = None
    item: Optional[List[Reference]] = field(default_factory=list)
    value_string: Optional[str] = None
    value_boolean: Optional[bool] = None
    value_date_time: Optional[str] = None
    value_date: Optional[str] = None
    value_coding: Optional[Coding] = None
    value_integer: Optional[int] = None
    value_attachment: Optional[Attachment] = None

@dataclass(kw_only=True)
class QuestionnaireResponseItem(BackboneElement):
    item: Optional[List[Reference]] = field(default_factory=list)
    text: Optional[str] = None
    answer: Optional[List[QuestionnaireResponseItemAnswer]] = field(default_factory=list)
    link_id: str
    definition: Optional[str] = None

@dataclass(kw_only=True)
class QuestionnaireResponse(DomainResource):
    questionnaire: Optional[str] = None
    encounter: Optional[Reference] = None
    item: Optional[List[QuestionnaireResponseItem]] = field(default_factory=list)
    source: Optional[Reference] = None
    author: Optional[Reference] = None
    status: str
    identifier: Optional[Identifier] = None
    based_on: Optional[List[Reference]] = field(default_factory=list)
    authored: Optional[str] = None
    part_of: Optional[List[Reference]] = field(default_factory=list)
    subject: Optional[Reference] = None